transaction-history-manager/src/main/java/com/qpay/transactionhistorymanager/service/impl/ReportGeneratorIText.java

```diff
@@ -0,0 +1,203 @@
+package com.qpay.transactionhistorymanager.service.impl;
+
+import com.itextpdf.io.image.ImageData;
+import com.itextpdf.io.image.ImageDataFactory;
+import com.itextpdf.layout.element.Image;
+import com.itextpdf.layout.element.Paragraph;
+import com.itextpdf.layout.element.Table;
+import com.itextpdf.layout.element.Text;
+import com.itextpdf.layout.properties.TextAlignment;
+import com.qpay.libs.FileManager;
+import com.qpay.libs.models.UserType;
+import com.qpay.transactionhistorymanager.client.UserClient;
+import com.qpay.transactionhistorymanager.model.TransactionType;
+import com.qpay.libs.models.ReportInfo;
+import com.qpay.transactionhistorymanager.model.entity.TransactionEntity;
+import com.qpay.transactionhistorymanager.repository.TransactionRepository;
+import com.qpay.transactionhistorymanager.service.ReportGeneratorService;
+import com.qpay.transactionhistorymanager.service.exception.SavePdfException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.net.MalformedURLException;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+import java.util.List;
+
+import static java.lang.String.format;
+
+@Service
+@RequiredArgsConstructor
+public class ReportGeneratorIText implements ReportGeneratorService {
+
+    private final TransactionRepository transactionRepository;
+
+    private final UserClient userClient;
+
+    private static final String QPAY_LOGO_IMAGE_PATH = "data/logo.png";
+
+    private static final String REPORT_PATH = "data/reports/";
+
+    private static final DateTimeFormatter DATE_PATTERN_dd_MM_yyyy = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+    private static final DateTimeFormatter DATE_PATTERN_dd_MM_yyyy_HH_mm = DateTimeFormatter.ofPattern("dd.MM.yyyy HH:mm");
+
+    private static final float LOGO_SCALE = 0.4F;
+
+    private static final float TITLE_MARGIN_TOP = 25;
+
+    private static final int TITLE_FONT_SIZE = 32;
+
+    private static final float SUBTITLE_MARGIN_TOP = 90;
+
+    private static final int SUBTITLE_FONT_SIZE = 14;
+
+    private static final float CLIENT_NAME_MARGIN_TOP = 15;
+
+    private static final float[] TABLE_COLUMN_WIDTHS = {50F, 150F, 100F, 150F, 250F};
+
+    private static final String REPORT_PERIOD = "For period from %s to %s";
+
+    public void generatePdfReport(final ReportInfo reportInfo) {
+        final var customerReportInfo = userClient.getCustomerReportInfo(reportInfo.userId());
+        final var path = generatePath(reportInfo.userId());
+        var totalReceived = BigDecimal.ZERO;
+        var totalSent = BigDecimal.ZERO;
+        final var transactionsTable = createTransactionsTable();
+        final List<TransactionEntity> transactionEntitiesInPeriod = getTransactionEntitiesInPeriod(reportInfo);
```

Comment:
```suggestion
        final var transactionEntitiesInPeriod = getTransactionEntitiesInPeriod(reportInfo);
```
this method returns list, you don't need to explicitly declare it here